### PR TITLE
Add missing checks for to_str calls in ENV specs

### PR DESCRIPTION
**Pull request by herwinw**
A number of specs did mention the calls to to_str, but did not test the full path.

### CODE REVIEW

File: core/env/delete_spec.rb [hunk 1]
```diff
@@ -41,6 +41,14 @@
     ENV["foo"].should == nil
   end
 
+  it "removes the variable coerced wit #to_str" do
```
Comment (seven1m) [inline core/env/delete_spec.rb:44]: wit -> with

Comment (herwinw) [reply core/env/delete_spec.rb]: I guess that was a witty comment

Comment (herwinw) [reply core/env/delete_spec.rb]: Resolved in the latest version

**Comment by andrykonchin** [comment]
Thank you!

**Pull request merged**